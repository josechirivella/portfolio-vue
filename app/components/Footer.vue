<template>
  <footer>
    <div class="copyright text-center p-4">
      &copy; {{ year }} Jose Chirivella. All rights reserved.
    </div>
  </footer>
</template>

<script lang="ts" setup>
const year = ref(new Date().getFullYear());
</script>
