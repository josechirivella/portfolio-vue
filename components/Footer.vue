<template>
  <footer>
    <div class="copyright text-center p-4">
      &copy; {{ year }} Jose Chirivella. All rights reserved.
    </div>
  </footer>
</template>

<script setup lang="ts">
const year = ref(new Date().getFullYear());
</script>
