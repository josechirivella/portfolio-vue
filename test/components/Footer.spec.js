import { beforeAll, describe, expect, test } from 'vitest';
import { shallowMount } from '@vue/test-utils';
import Footer from '@/components/Footer';

describe('Footer', () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallowMount(Footer, {
      data() {
        return {
          year: new Date().getFullYear(),
        };
      },
    });
  });
  test('should render', () => {
    expect(wrapper.vm).toBeTruthy();
    expect(wrapper.vm.$el).toMatchSnapshot();
  });
  test('should set the current year', () => {
    const year = new Date().getFullYear();
    expect(wrapper.text()).toContain(year.toString());
  });
});
