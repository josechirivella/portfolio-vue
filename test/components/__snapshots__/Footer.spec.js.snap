// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer > should render 1`] = `
<footer>
  <div
    class="copyright text-center p-4"
  >
     Â© 2025 Jose Chirivella. All rights reserved. 
  </div>
</footer>
`;
